eclair-core/src/main/scala/fr/acinq/eclair/db/psql/PsqlChannelsDb.scala

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2019 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.db.psql
+
+import fr.acinq.bitcoin.ByteVector32
+import fr.acinq.eclair.CltvExpiry
+import fr.acinq.eclair.channel.HasCommitments
+import fr.acinq.eclair.db.ChannelsDb
+import fr.acinq.eclair.db.psql.PsqlUtils.DatabaseLock
+import fr.acinq.eclair.wire.ChannelCodecs.stateDataCodec
+import grizzled.slf4j.Logging
+import javax.sql.DataSource
+
+import scala.collection.immutable.Queue
+
+class PsqlChannelsDb(implicit ds: DataSource, lock: DatabaseLock) extends ChannelsDb with Logging {
+
+  import PsqlUtils.ExtendedResultSet._
+  import PsqlUtils._
+  import lock._
+
+  val DB_NAME = "channels"
+  val CURRENT_VERSION = 2
+
+  inTransaction { psql =>
+    using(psql.createStatement()) { statement =>
+      getVersion(statement, DB_NAME, CURRENT_VERSION) match {
+        case CURRENT_VERSION =>
+          statement.executeUpdate("CREATE TABLE IF NOT EXISTS local_channels (channel_id TEXT NOT NULL PRIMARY KEY, data BYTEA NOT NULL, is_closed BOOLEAN NOT NULL DEFAULT FALSE)")
+          statement.executeUpdate("CREATE TABLE IF NOT EXISTS htlc_infos (channel_id TEXT NOT NULL, commitment_number TEXT NOT NULL, payment_hash TEXT NOT NULL, cltv_expiry BIGINT NOT NULL, FOREIGN KEY(channel_id) REFERENCES local_channels(channel_id))")
```

Comment:
Do we really need to user `BIGINT` data type for `cltv_expiry`?

Reply:
Ditto.